sktime/clustering/base/base.py

```diff
@@ -0,0 +1,198 @@
+
+    def _predict(self, X: Numpy_Array) -> Numpy_Array:
+        raise NotImplementedError("abstract method")
+
+    def _check_params(self, X: Numpy_Array):
+        """
+        Method used to check the parameters passed
+
+        Parameters
+        ----------
+        X: Numpy_Array
+            Dataset to be validate parameters against
+        """
+        return
+
+
+class ClusterMixin:
+    def fit_predict(self, X: Numpy_Or_DF) -> Numpy_Array:
+        """
+        Method that calls fit and then returns a prediction
+        for the value of X
+
+        Parameters
+        ----------
+        X: Numpy array or Dataframe
+            sktime data_frame or numpy array containing the values
+            to perform clustering algorithm on
+        """
+        self.fit(X)
+        return self.predict(X)
+
+
+class BaseClusterCenterInitializer:
+    def __init__(self, data_set: Numpy_Array, n_centers: int):
+        """
+        Constructor for BaseClusterCenterInitializer
+
+        Parameters
+        ----------
+        data_set: Numpy_Array
+            Numpy_Array that is the dataset to calculate the centers from
+
+        n_centers: int
+            Number of centers to be created
+        """
+        self.data_set = data_set
+        self.n_centers = n_centers
+
+    def initialize_centers(self) -> Numpy_Array:
+        """
+        Method used to initialise centers
+
+        Returns
+        -------
+        Data_Frame
+            sktime dataframe containing the starting
+            centers
+        """
+        raise NotImplementedError("abstract method")
+
+
+class BaseClusterAverage:
+    def __init__(self, series: Numpy_Array, n_iterations: int = 10):
+        """
+        Constructor for BaseClusterAverage
+
+        Parameters
+        ----------
+        series: Numpy_Array
+            Set of series to generate a average from
+        """
+        self.n_iterations = n_iterations
+        self.series = series
+
+    def average(self) -> Numpy_Array:
+        """
+        Method called to find the average for a distance metric
+
+        Returns
+        -------
+        Numpy_Array
+            Created array denoting the average
+
+        n_iterations: int
+            Number of iterations to refine the average
+        """
+        raise NotImplementedError("abstract method")
+
+
+class BaseApproximate:
+    def __init__(self, series: Numpy_Array):
+        """
+        Constructor for BaseAprroximate
+
+        Parameters
+        ----------
+        series: Numpy_Array
+            series to perform approximation on
+
+        """
+        self.series = series
+
+    def approximate(self) -> int:
+        """
+        Method called to get the approximation
+
+        Returns
+        -------
+        int
+            Index position of the approximation in the series
+        """
+        raise NotImplementedError("abstract method")
+
+
+Init_Algo = Union[str, BaseClusterCenterInitializer]
```

Comment:
I suggest to have a separate file for these (e.g. `sktime/clustering/base/_typing.py`) or move them to the top of the file